Documentation/build-parameters.rst

```diff
@@ -0,0 +1,105 @@
+..
+  SPDX-FileCopyrightText: 2024 SeisSol Group
+
+  SPDX-License-Identifier: BSD-3-Clause
+
+.. _build_parameters:
+
+Build Parameter Reference
+=========================
+
+**For Developers:** if you add a CMake parameter, make sure to document it here.
```

Comment:
I would remove this sentence. it seems clear that if you add new features you should document them.

Reply:
I'd actually see it differently here: it can never be clear enough. :)
Unless it's placed better somewhere else, of course.